dockstore-webservice/src/main/java/io/swagger/api/impl/ToolsApiServiceImpl.java

```diff
@@ -393,16 +398,14 @@ public Response toolsGet(String id, String alias, String registry, String organi
             handleParameter(registry, "registry", filters);
             handleParameter(limit.toString(), "limit", filters);
 
-            int port = config.getExternalConfig().getPort() == null ? -1 : Integer.parseInt(config.getExternalConfig().getPort());
-
             if (offsetInteger + 1 < pagedResults.size()) {
                 URI nextPageURI = new URI(config.getExternalConfig().getScheme(), null, config.getExternalConfig().getHostname(), port,
-                    config.getExternalConfig().getBasePath() + DockstoreWebserviceApplication.GA4GH_API_PATH + "/tools",
+                    ObjectUtils.firstNonNull(config.getExternalConfig().getBasePath(), "") + DockstoreWebserviceApplication.GA4GH_API_PATH + "/tools",
                     Joiner.on('&').join(filters) + "&offset=" + (offsetInteger + 1), null).normalize();
                 responseBuilder.header("next_page", nextPageURI.toURL().toString());
             }
             URI lastPageURI = new URI(config.getExternalConfig().getScheme(), null, config.getExternalConfig().getHostname(), port,
-                config.getExternalConfig().getBasePath() + DockstoreWebserviceApplication.GA4GH_API_PATH + "/tools",
+                ObjectUtils.firstNonNull(config.getExternalConfig().getBasePath(), "") + DockstoreWebserviceApplication.GA4GH_API_PATH + "/tools",
```

Comment:
I think this code is a dupe of lines 402-403, except for the value of the offset query parameter; you could have a common method that these both call.